pulsar-client-cpp/include/pulsar/Auth.h

```diff
@@ -26,9 +26,27 @@
 #pragma GCC visibility push(default)
 
 namespace pulsar {
-    
-    typedef std::map<std::string, std::string> ParamMap;
-    
+
+    // for compatibility with previous version
+    enum AuthType {
```

Comment:
Can we mark the compatibility method as deprecated? This way we can remove them once all the users updated the code.